source/Missions/Services/Agents/Handlers/WeaponPickupHandler.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Autofac;
+using Common.Messaging;
+using Common.Network;
+using GameInterface.Serialization;
+using HarmonyLib;
+using Missions.Services.Agents.Messages;
+using Missions.Services.Network;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using TaleWorlds.Core;
+using TaleWorlds.MountAndBlade;
+
+namespace Missions.Services.Agents.Handlers
+{
+    public class WeaponPickupHandler
+    {
+
+        public WeaponPickupHandler()
+        {
+            ContainerBuilder builder = new ContainerBuilder();
+            builder.RegisterModule<MissionModule>();
+            container = builder.Build();
+
+            binaryPackageFactory = container.Resolve<IBinaryPackageFactory>();
```

Comment:
This is an anti-pattern, do the following instead
```suggestion
        public WeaponPickupHandler(IBinaryPackageFactory binaryPackageFactory)
        {
            this.binaryPackageFactory = binaryPackageFactory;
```